src/mod_muc_db_mnesia.erl

```diff
@@ -0,0 +1,182 @@
+%%%----------------------------------------------------------------------
+%%% Based on file mod_muc.
+%%%
+%%% Original header and copyright notice:
+%%%
+%%% Author  : Alexey Shchepin <alexey@process-one.net>
+%%% Purpose : MUC support (XEP-0045)
+%%%
+%%%
+%%% ejabberd, Copyright (C) 2002-2011   ProcessOne
+%%%
+%%% This program is free software; you can redistribute it and/or
+%%% modify it under the terms of the GNU General Public License as
+%%% published by the Free Software Foundation; either version 2 of the
+%%% License, or (at your option) any later version.
+%%%
+%%% This program is distributed in the hope that it will be useful,
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License
+%%% along with this program; if not, write to the Free Software
+%%% Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+%%% 02111-1307 USA
+%%%
+%%%----------------------------------------------------------------------
+
+-module(mod_muc_db_mnesia).
+-behaviour(mod_muc_db).
+-export([init/2,
+         store_room/4,
+         restore_room/3,
+         forget_room/3,
+         get_rooms/2,
+         can_use_nick/4,
+         get_nick/3,
+         set_nick/4,
+         unset_nick/3]).
+
+-include("mongoose.hrl").
+-include("jlib.hrl").
+-include("mod_muc.hrl").
+
+init(_Host, _Opts) ->
+    mnesia:create_table(muc_room,
+                        [{disc_copies, [node()]},
+                         {attributes, record_info(fields, muc_room)}]),
+    mnesia:create_table(muc_registered,
+                        [{disc_copies, [node()]},
+                         {attributes, record_info(fields, muc_registered)}]),
+    mnesia:add_table_copy(muc_room, node(), disc_copies),
+    mnesia:add_table_copy(muc_registered, node(), disc_copies),
+    mnesia:add_table_index(muc_registered, nick),
+    ok.
+
+-spec store_room(ejabberd:server(), ejabberd:server(), mod_muc:room(), list())
+            -> {'aborted', _} | {'atomic', _}.
+store_room(_ServerHost, MucHost, RoomName, Opts) ->
+    F = fun() ->
+                mnesia:write(#muc_room{name_host = {RoomName, MucHost},
+                                       opts = Opts})
+        end,
+    Result = mnesia:transaction(F),
+    case Result of
+        {atomic, _} ->
+            ok;
+        _ ->
+            ?ERROR_MSG("issue=store_room_failed room=~ts", [RoomName])
+    end,
+    Result.
+
+
+restore_room(_ServerHost, MucHost, RoomName) ->
+    try mnesia:dirty_read(muc_room, {RoomName, MucHost}) of
+        [#muc_room{opts = Opts}] ->
+            {ok, Opts};
+        [] ->
+            {error, room_not_found};
+        Other ->
+            {error, Other}
+        catch Class:Reason ->
```

Comment:
You seem to consistently ident `catch` in this PR as if it were a clause inside `try`. This is inconsistent with how we (and [most of the Erlang world](http://erlang.org/doc/reference_manual/expressions.html#try), including indentation tools I use) do it everywhere else.

Reply:
oh, no, I don't like the world.